packages/next-server/server/next-server.ts

```diff
@@ -319,7 +319,15 @@ export default class Server {
     query: ParsedUrlQuery = {},
     opts: any,
   ) {
+    // try serving a static AMP version first
+    if (query.amp) {
+      try {
+        const result = await loadComponents(this.distDir, this.buildId, (pathname === '/' ? '/index' : pathname) + '.amp')
+        if (typeof result.Component === 'string') return result.Component
+      } catch (_) {}
```

Comment:
This should handle the errors coming from the execution.

Reply:
Updated